README.md

```diff
@@ -9,19 +9,27 @@ The current focus is on Docker and containerd, contributions are welcome for oth
-It reloads the unit files from disk and reevaluates `multi-user.target`, `sockets.target`, and `timers.target`, making sure your enabled systemd units run.
-This service is bound to `systemd-sysext.service` which activates the sysext images on boot.
-At runtime executing `systemctl restart systemd-sysext` will reload the sysext images and start the services.
+This helper service is bound to `systemd-sysext.service` which activates the sysext images on boot.
+Currently it reloads the unit files from disk and reevaluates `multi-user.target`, `sockets.target`, and `timers.target`, making sure your enabled systemd units run.
+In the future `systemd-sysext` will only reload the unit files when this is upstream behavior (the current upstream behavior is to do nothing and leave it to the user).
+That means you will have to use `Upholds=` drop-ins for the target units to start your units.
+At runtime executing `systemctl restart systemd-sysext ensure-sysext` will reload the sysext images and start the services.
 A manual `systemd-sysext refresh` is not recommended.
 
 The compatibility mechanism of sysext images requires a metadata file in the image under `usr/lib/extension-release.d/extension-release.NAME`.
-It has to contain a matching OS `ID`, and either a matching `VERSION` or `SYSEXT_LEVEL`.
-Since the rapid release cycle and automatic updates of Flatcar Container Linux make it hard to rely on particular OS libraries by specifying a dependency of the sysext image to the OS version, it is not recommended to match by `VERSION`.
+It has to contain a matching OS `ID`, and either a matching `VERSION_ID` or `SYSEXT_LEVEL`.
+Since the rapid release cycle and automatic updates of Flatcar Container Linux make it hard to rely on particular OS libraries by specifying a dependency of the sysext image to the OS version, it is not recommended to match by `VERSION_ID`.
 Instead, Flatcar defined the `SYSEXT_LEVEL` value `1.0` to match for.
 The sysext image should only include static binaries.
 
 Inside the image, binaries should be placed under `usr/bin/` and systemd units under `usr/lib/systemd/system/`.
-To enable systemd units, symlinks should be included in the image itself in the same way as systemd would normally generate them when enabling the units, e.g., `sockets.target.wants/my.socket` → `../my.socket`.
+To enable systemd units, while it is currently supported to place symlinks in the image itself in the same way as systemd would normally generate them when enabling the units, e.g., `sockets.target.wants/my.socket` → `../my.socket`, this is not recommended.
+The recommended way is to ship drop-ins for the target units that start your unit, e.g., `/usr/lib/systemd/system/sockets.target.d/10-docker-socket.conf` would contain (similar for `multi-user.target` and a `.service` unit):
+
+```
+[Unit]
+Upholds=docker.socket
+```
 
```

Comment:
Here I'd add something like:

```
This can be done also for services, so for `docker.service` started by `multi-user.target`, the drop-in would reside in `/usr/lib/systemd/system/multi-user.target.d/10-docker-service.conf` and it would have `Upholds=docker.service` line.
```

Reply:
Thanks